### PR TITLE
Generate TS type definitions from jsdoc

### PR DESCRIPTION
**Pull request by stereobooster**
WIP blocked by https://github.com/englercj/tsd-jsdoc/issues/35

### CODE REVIEW

File: out/types.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ *   p { background-color: red; }
+ * You can then instead get:
+ *   /*! important 1 *\/
+ *   /*! important 2 *\/
+ *   p { color: red; background-color: red; }
+ * @param {string} css
+ * @return {string}
+ */
+declare function collectImportantComments(css: string): string;
+
```
Comment (peterbe) [inline out/types.d.ts]: What *is* this file?

Comment (stereobooster) [reply out/types.d.ts]: this is type definition - if anybody will use this project as library, one will get typescript definitions e.g. you can use it in TypeScript or get intellisense in Code etc.

**Comment by stereobooster** [comment]
> I'm guessing you can figure out the conflict. Was it to do with the new option to the minimize function?

Yes, but anyway this is not ready for merge. It doesn't work as expected.

**Comment by peterbe** [comment]
Exciting! I don't even know how to test that once it's done but it sounds exciting for sure.

**Comment by stereobooster** [comment]
Would be fixed as soon as this will be fixed https://github.com/Microsoft/TypeScript/issues/7546

**Pull request closed without merging**